File: java/src/moonaframework/base/MoonaHandlingException.java
package moonaframework.base;

public class MoonaHandlingException extends RuntimeException {

	private static final long serialVersionUID = -1L;
	
	public static final MoonaHandlingException CRITICAL = new MoonaHandlingException("This should not happen...");
	
	public MoonaHandlingException(String message, Throwable cause) {
		super(message, cause);
	}
	
	public MoonaHandlingException(String message) {
		super(message);
	}
	
	public MoonaHandlingException() {
		super("Something went wrong with Moona...");
	}
}

File: cpp/base/moonahandlingexception.cpp
#include "moonahandlingexception.hpp"

namespace moona {
    
    MoonaHandlingException::MoonaHandlingException() : Exception("Something went wrong with Moona.") {
    }

    MoonaHandlingException::MoonaHandlingException(const char* message) : Exception(message) {
    }

    MoonaHandlingException::~MoonaHandlingException() {
    }
}